sphinx_gallery/clean.py

```diff
@@ -0,0 +1,29 @@
+from argparse import ArgumentParser
+import os.path
+import shutil
+import subprocess
+import sys
+
+
+def main():
+    parser = ArgumentParser(
+        description=
+        "Clean intermediate files generated by sphinx-gallery, and call "
+        "'python -msphinx -M clean' with the same arguments.")
+    parser.add_argument("sourcedir")
+    parser.add_argument("builddir")
+    args, _ = parser.parse_known_args()
+    d = {}
+    with open(os.path.join(args.sourcedir, "conf.py")) as file:
+        exec(file.read(), d)
+    sg_conf = d["sphinx_gallery_conf"]
+    for path in sg_conf["gallery_dirs"]:
+        shutil.rmtree(os.path.join(args.sourcedir, path), ignore_errors=True)
+    shutil.rmtree(os.path.join(args.sourcedir, sg_conf["backreferences_dir"]),
+                  ignore_errors=True)
+    subprocess.check_call(
+        [sys.executable, "-msphinx", "-M", "clean"] + sys.argv[1:])
```

Comment:
why `argv` and not the directories that where just parsed?

Reply:
Because there can be a bunch of other args that are meaningful to sphinx:
```
optional arguments:
  -h, --help        show this help message and exit
  --version         show program's version number and exit

general options:
  -b BUILDER        builder to use (default: html)
  -a                write all files (default: only write new and changed
                    files)
  -E                don't use a saved environment, always read all files
  -d PATH           path for the cached environment and doctree files
                    (default: OUTPUTDIR/.doctrees)
  -j N              build in parallel with N processes where possible (special
                    value "auto" will set N to cpu-count)

build configuration options:
  -c PATH           path where configuration file (conf.py) is located
                    (default: same as SOURCEDIR)
  -C                use no config file at all, only -D options
  -D setting=value  override a setting in configuration file
  -A name=value     pass a value into HTML templates
  -t TAG            define tag: include "only" blocks with TAG
  -n                nit-picky mode, warn about all missing references

console output options:
  -v                increase verbosity (can be repeated)
  -q                no output on stdout, just warnings on stderr
  -Q                no output at all, not even warnings
  --color           do emit colored output (default: auto-detect)
  -N, --no-color    do not emit colored output (default: auto-detect)
  -w FILE           write warnings (and errors) to given file
  -W                turn warnings into errors
  -T                show full traceback on exception
  -P                run Pdb on exception
```
most of them probably don't matter for cleaning, but we may as well make everything work even if sphinx decides to start pulling stuff from these args when cleaning too.